app/Jobs/GenerateTicket.php

```diff
@@ -2,93 +2,54 @@
 
     /**
      * Create a new job instance.
      *
-     * @return void
+     * @param  Order  $order
+     * @param  Attendee  $attendee
      */
-    public function __construct($reference)
+    public function __construct(Order $order, Attendee $attendee = null)
     {
-        Log::info("Generating ticket: #" . $reference);
-        $this->reference = $reference;
-        $this->order_reference = explode("-", $reference)[0];
-        if (strpos($reference, "-")) {
-            $this->attendee_reference_index = explode("-", $reference)[1];
-        }
+        $this->order = $order;
+        $this->attendee = $attendee;
     }
 
     /**
      * Execute the job.
      *
      * @return void
      */
-    public function handle()
-    {
-
-        $file_name = $this->reference;
-        $file_path = public_path(config('attendize.event_pdf_tickets_path')) . '/' . $file_name;
-        $file_with_ext = $file_path . ".pdf";
-
-        if (file_exists($file_with_ext)) {
-            Log::info("Use ticket from cache: " . $file_with_ext);
-            return;
-        }
-
-        $order = Order::where('order_reference', $this->order_reference)->first();
-        Log::info($order);
-        $event = $order->event;
-
-        $query = $order->attendees();
-        if ($this->isAttendeeTicket()) {
-            $query = $query->where('reference_index', '=', $this->attendee_reference_index);
-        }
-        $attendees = $query->get();
-
-        $image_path = $event->organiser->full_logo_path;
-        $images = [];
-        $imgs = $order->event->images;
-        foreach ($imgs as $img) {
-            $images[] = base64_encode(file_get_contents(public_path($img->image_path)));
-        }
-
-        $data = [
-            'order'     => $order,
-            'event'     => $event,
-            'attendees' => $attendees,
-            'css'       => file_get_contents(public_path('assets/stylesheet/ticket.css')),
-            'image'     => base64_encode(file_get_contents(public_path($image_path))),
-            'images'    => $images,
-        ];
-        try {
-            PDF::setOutputMode('F'); // force to file
-            PDF::html('Public.ViewEvent.Partials.PDFTicket', $data, $file_path);
-            Log::info("Ticket generated!");
-        } catch(\Exception $e) {
-            Log::error("Error generating ticket. This can be due to permissions on vendor/nitmedia/wkhtml2pdf/src/Nitmedia/Wkhtml2pdf/lib. This folder requires write and execute permissions for the web user");
-            Log::error("Error message. " . $e->getMessage());
-            Log::error("Error stack trace" . $e->getTraceAsString());
-            $this->fail($e);
-        }
-
-    }
-
-    private function isAttendeeTicket()
+    public function handle(): void
     {
-        return ($this->attendee_reference_index != null);
+        $this->pdf_file = TicketGenerator::createPDFTicket($this->order, $this->attendee);
+        $this->pdf_file->error ?: $this->fail();
```

Comment:
Hi, just found this small mistake while trying to use this PR.
Actually it means that `GenerateTicket` will fail if `$this->pdf_file->error === false` which is probably not what was intended ;-)